Mathlib/Lean/Meta.lean

```diff
@@ -41,6 +41,11 @@ where
       else
         return e.hasFVar
 
+/-- Add the hypothesis `h : t`, given `v : t`, and return the new `FVarId`. -/
+def note (h : Name := .anonymous) (t : Option Expr) (v : Expr) (g : MVarId) :
+    MetaM (FVarId × MVarId) := do
+  (← g.assert h (t.getD (← inferType v)) v).intro1P
```

Comment:
The `getD` here doesn't work, you need `getDM` as mentioned [on zulip](https://leanprover.zulipchat.com/#narrow/stream/270676-lean4/topic/.5BRFC.5D.20refine_struct.20functionality.20via.20new.20.3F.2E.2E.20syntax/near/314351174)

Reply:
Fixed, thanks.